scripts/timeseries/importGeojson.js

```diff
@@ -19,22 +25,40 @@ const allModes = [
   'UNKNOWN'
 ]
 
-const N_DOCS = 75
+const arrayHasSameValue = arr => {
+  return uniqWith(arr, isEqual).length === 1
+}
+
+const getRandomValueFromArray = (arr, exclude) => {
+  const randIdx = Math.floor(Math.random() * arr.length)
+  const selectedValue =
+    exclude && isEqual(exclude, arr[randIdx]) && !arrayHasSameValue(arr)
+      ? getRandomValueFromArray(arr, exclude)
+      : arr[randIdx]
+
+  return selectedValue
+}
 
 const createTrips = () => {
   const trips = []
 
-  for (let i = 0; i < N_DOCS; i++) {
-    const randomDoc = GEODOCS[Math.floor(Math.random() * GEODOCS.length)]
+  for (let idx = 0; idx < N_DOCS; idx++) {
+    const randomDoc = cloneDeep(getRandomValueFromArray(GEODOCS))
     const cozyMetadata = { sourceAccount: ACCOUNTS[0]._id }
 
     randomDoc.series[0].features.forEach(featColl => {
       if (featColl.type === 'FeatureCollection') {
-        const mode = allModes[Math.floor(Math.random() * allModes.length)]
+        const mode = getRandomValueFromArray(allModes)
         featColl.features[0].properties.sensed_mode = `PredictedModeTypes.${mode}`
       }
     })
 
+    const startName = getRandomValueFromArray(STREETS)
+    const endName = getRandomValueFromArray(STREETS, startName)
```

Comment:
Since the name is actually false compared to the GPS values, I wouldn't mind having the occasionally same start/end name, as it is actually a valid trip to come back to the starting point.

Reply:
Indeed, it remains coherent